### PR TITLE
Disable the use of user namespaces.

### PR DESCRIPTION
**Pull request by adelton**
#### Description:

- Disable the use of user namespaces.

#### Rationale:

- User namespaces are used primarily for Linux containers. The value 0 effectively disallows the use of user namespaces.
- This rule is optional because it only applies to machines where containers are not deployed; hence no sysctl_values.csv entries.

### CODE REVIEW

File: linux_os/guide/system/permissions/restrictions/sysctl_user_max_user_namespaces/rule.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    User namespaces are used primarily for Linux containers. The value 0
+    effectively disallows the use of user namespaces.
+
+severity: low
+
+identifiers:
+    cce@rhel8: 81005-1
+
+references:
+    ospp: FMT_SMF_EXT.1
```
Comment (shawndwells) [inline linux_os/guide/system/permissions/restrictions/sysctl_user_max_user_namespaces/rule.yml]: What do you think about mapping to NIST 800-53, SC-39? https://nvd.nist.gov/800-53/Rev4/control/SC-39

File: linux_os/guide/system/permissions/restrictions/sysctl_user_max_user_namespaces/rule.yml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+title: 'Disable the use of user namespaces'
+
+description: |-
+    {{{ describe_sysctl_option_value(sysctl="user.max_user_namespaces", value="0") }}}
+    When containers are deployed on the machine, the value should be set
+    to large non-zero value.
+
+rationale: |-
+    User namespaces are used primarily for Linux containers. The value 0
+    effectively disallows the use of user namespaces.
```
Comment (shawndwells) [inline linux_os/guide/system/permissions/restrictions/sysctl_user_max_user_namespaces/rule.yml]:
```suggestion
   disallows the use of user namespaces.
```

Comment (shawndwells) [reply linux_os/guide/system/permissions/restrictions/sysctl_user_max_user_namespaces/rule.yml]: What is the nuance of saying "effectively disables" vs "disables"?

File: linux_os/guide/system/permissions/restrictions/sysctl_user_max_user_namespaces/rule.yml [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
```
Comment (shawndwells) [inline linux_os/guide/system/permissions/restrictions/sysctl_user_max_user_namespaces/rule.yml]:
```suggestion
warnings:
    - general: |-
       This configuration baseline was created to deploy the base operating system for general purpose
       workloads. When the operating system is configured for certain purposes, such as to host Linux Containers,
       it is expected that <tt>user.max_user_namespaces</tt> will be enabled.
```

Comment (shawndwells) [reply linux_os/guide/system/permissions/restrictions/sysctl_user_max_user_namespaces/rule.yml]: Lets give administrators and auditors an understanding of when to expect user.max_user_namespaces to be enabled.

File: linux_os/guide/system/permissions/restrictions/sysctl_user_max_user_namespaces/rule.yml (hunk 1 above)
Comment (adelton) [reply linux_os/guide/system/permissions/restrictions/sysctl_user_max_user_namespaces/rule.yml]: I've added `nist: SC-39` in d518ade748d0ca02854f57d51f6f3174d4394823.

File: linux_os/guide/system/permissions/restrictions/sysctl_user_max_user_namespaces/rule.yml (hunk 2 above)
Comment (adelton) [reply linux_os/guide/system/permissions/restrictions/sysctl_user_max_user_namespaces/rule.yml]: Removed.

File: linux_os/guide/system/permissions/restrictions/sysctl_user_max_user_namespaces/rule.yml (hunk 3 above)
Comment (adelton) [reply linux_os/guide/system/permissions/restrictions/sysctl_user_max_user_namespaces/rule.yml]: Added in d518ade748d0ca02854f57d51f6f3174d4394823.

**Comment by adelton** [comment]
Addressed comments, rebased on master -> d518ade748d0ca02854f57d51f6f3174d4394823.

**Comment by yuumasato** [comment]
@adelton Unfortunately, another PR that was merged caused conflicts with the CCE file.

**Comment by adelton** [comment]
Rebased on master -> 885d55a91c1788cb07f0259d6a328a3047b8da33.

**Comment by yuumasato** [comment]
Thanks!

**Pull request merged**